src/vellum/workflows/runner/runner.py

```diff
@@ -409,6 +411,21 @@ def _handle_invoked_ports(
                     self._concurrency_queue.put((next_state, edge.to_node, invoked_by))
                 else:
                     self._run_node_if_ready(next_state, edge.to_node, invoked_by)
+                if invoked_by:
+                    previous_node = state.meta.node_execution_cache.__node_execution_lookup__.get(invoked_by)
+                    if previous_node:
+                        previous_node_fork_id = state.meta.node_execution_cache.__node_to_fork_id__.get(previous_node)
+                        if previous_node_fork_id:
+                            state.meta.node_execution_cache.__node_to_fork_id__[edge.to_node] = previous_node_fork_id
+                            # Remove previous node fork in __node_to_fork_id__
+                            del state.meta.node_execution_cache.__node_to_fork_id__[previous_node]
+
+            # If there are multiple edges, we need to create a fork ID
+            if len(nodes_to_fork) > 1:
+                fork_id = uuid4()
+                state.meta.node_execution_cache.__all_fork_ids__.add(fork_id)
+                for node in nodes_to_fork:
+                    state.meta.node_execution_cache.__node_to_fork_id__[node] = fork_id
```

Comment:
Let's sync up tm - would love to understand more about the need for this fork id

In general, I would love to find a way to keep logic outside of workflow runner and contained to the AWAIT_ANY case within the Node's Trigger